tests/server/test_plugins.py

```diff
@@ -43,79 +149,355 @@ def test_example_config(self):
+
+        delta = (plugin.next_run - datetime.datetime.now())
+        assert round(delta.total_seconds(), 0) == plugin.params['Interval']
+
+    def test_force_to_run_plugin_without_interval_parameter(self):
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=self.test_params_default)
+        plugin.forced = True
+        plugin.run()
+        time.sleep(0.5)
+        assert plugin.current_run
+        assert type(plugin.current_run).__name__ == 'PluginWorker'
+        assert not plugin.next_run
+        assert plugin.forced
+
+    def test_force_to_run_plugin_with_interval_parameter(self):
+        # should not schedule next_run
+        params = dict(self.test_params_default, **{'Interval': 2})
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=params)
+        n = plugin.next_run
+        plugin.forced = True
+
+        plugin.run()
+        time.sleep(0.5)
+        assert plugin.current_run
+        assert type(plugin.current_run).__name__ == 'PluginWorker'
+        assert n == plugin.next_run
+        assert plugin.forced
+
+    def test_run_plugin_with_interval(self):
+        params = dict(self.test_params_default, **{'Interval': 2})
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=params)
+        time.sleep(2.5)
+        assert not plugin.current_run
+        plugin.run()
+        time.sleep(0.5)
+        assert plugin.current_run
+        assert type(plugin.current_run).__name__ == 'PluginWorker'
+
+    def test_run_plugin_with_interval_should_schedule_next_run(self):
+        params = dict(self.test_params_default, **{'Interval': 2})
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=params)
+        time.sleep(2.5)
+        plugin.run()
+        delta = (plugin.next_run - datetime.datetime.now())
+        assert round(delta.total_seconds(), 0) == plugin.params['Interval']
+
+    def test_plugin_should_not_be_run_without_interval_parameter(self):
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=self.test_params_default)
+        assert not plugin.next_run
+        plugin.forced = False
+        plugin.run()
+        time.sleep(0.5)
+        assert not plugin.current_run
+
+    def test_interval_run_plugin_and_collect_new_result(self):
+        params = dict(self.test_params_default, **{'Interval': 2})
+        plugin = server_plugins.Plugin(name=self.test_plugin_name,
+                                       params=params)
+        expected_result = {'status': 'OK', 'forced': False}
+        time.sleep(2.5)
+        plugin.run()
+        time.sleep(0.5)
+
+        assert not plugin.get_last_result()
+        plugin.collect_new_result()
+        assert cmp_dict(plugin.get_last_result(), expected_result) == 2
```

Comment:
this is not very readable, I'd preffer
```python
result = plugin.get_last_result()
assert 'status' in result and result['status'] == 'OK'
assert 'forced' in result and not result['forced']
```
This is clear what it means, calling a function and comparing result to a magic constant is not...